src/ui/ProcedureSpecEditor.as

```diff
@@ -296,6 +355,24 @@ public class ProcedureSpecEditor extends Sprite {
 		return result;
 	}
 
+	private function makeMenuArg(type:String):BlockArg {
+		var result:BlockArg = new BlockArg('m', 0xFFFFFF, true, type, false, true);
+		result.setArgValue(unusedArgName('menu'));
+		return result;
+	}
+
+	private function makeNumberMenuArg(type:String):BlockArg {
+		var result:BlockArg = new BlockArg('d', 0xFFFFFF, true, type);
+		result.setArgValue(unusedArgName('menu'));
+		return result;
+	}
+
+	private function chooseMenuType(action:Function):void {
+		var m:Menu = new Menu(function(result:String):void { appendObj(action(result)); });
```

Comment:
Is this how the other parts of the project do it..?

I'd rather it be something like this:

```as3
function useMenuType(selection:String) {
	appendObj(action(selection));
}

var m:Menu = new Menu(selectedCb);
```